src/parquet/reader-test.cc

```diff
@@ -84,36 +84,56 @@ TEST_F(TestAllTypesPlain, TestBatchRead) {
   ASSERT_FALSE(col->HasNext());
 }
 
+void verifyScanner(Int32Scanner& scanner) {
+  int32_t val;
+  bool is_null;
+  for (size_t i = 0; i < 8; ++i) {
+    ASSERT_TRUE(scanner.HasNext());
+    ASSERT_TRUE(scanner.NextValue(&val, &is_null));
+    ASSERT_FALSE(is_null);
+  }
+  ASSERT_FALSE(scanner.HasNext());
+  ASSERT_FALSE(scanner.NextValue(&val, &is_null));
+}
```

Comment:
It's a bummer that this only works with REQUIRED data and files with 8 values. I feel like we're digging the technical debt hole deeper by not taking this opportunity to write unit tests vs increasing our reliances on binary files. Thoughts?